### PR TITLE
[SPARK-32310][ML][PySpark] ML params default value parity in feature and tuning

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
set params default values in trait Params for feature and tuning in both Scala and Python. 

### Why are the changes needed?
Make ML has the same default param values between estimator and its corresponding transformer, and also between Scala and Python.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing and modified tests

### CODE REVIEW

**Comment by huaxingao** [comment]
retest this please

**Comment by huaxingao** [comment]
cc @srowen @viirya @zhengruifeng

**Comment by srowen** [comment]
Looks good if there are no API or behavior changes (that are not bug fixes).

**Review by zhengruifeng — APPROVED** [review]
I reviewed twice, it LGTM if related places were all changed.

**Comment by huaxingao** [comment]
retest this please

File: python/pyspark/ml/regression.py [hunk 1]
```diff
@@ -2391,7 +2404,7 @@ def __repr__(self):
 
 
 class _FactorizationMachinesParams(_PredictorParams, HasMaxIter, HasStepSize, HasTol,
-                                   HasSolver, HasSeed, HasFitIntercept, HasRegParam):
+                                   HasSolver, HasSeed, HasFitIntercept, HasRegParam, HasWeightCol):
```
Comment (viirya) [inline python/pyspark/ml/regression.py:2407]: Why add `HasWeightCol`? Seems you don't add a default of weight col?

**Review by viirya — COMMENTED** [review]
Looks good except for one question.

File: python/pyspark/ml/regression.py (hunk 1 above)
Comment (huaxingao) [reply python/pyspark/ml/regression.py:2407]: @viirya Thanks for reviewing. I added ```HasWeightCol``` in ```FactorizationMachinesParams``` in Scala in https://github.com/apache/spark/pull/28960 when I added ```FMClassificationSummary```.  ```ClassificationSummary``` needs weightCol, https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/ClassificationSummary.scala#L48. ```FMClassifier/Regressor``` doesn't support instance weight yet, so I didn't add ```setWeightCol``` or set default value in ```FactorizationMachinesParams```.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
I think you can merge when ready

**Comment by huaxingao** [comment]
Merged to master. Thank you all for reviewing!

**Pull request closed without merging**